Document making cross-forest trust work for Satellite (#3231)
theforeman/foreman-documentation: guides/common/modules/proc_configuring-projectserver-to-use-cross-forest-trust.adoc

> [id="Configuring-a-cross-forest-trust-between-{FreeIPA-context}-and-Active-Directory-for-{Project}_{context}"]

Author: Seriously, how do you notice these things? Is it a superpower or is it a script that you run? (Also, fixed.)

= Configuring a cross-forest trust between {FreeIPA} and Active Directory for {Project}

> When a cross-forest trust is configured in your {FreeIPA} deployment, configure host-based access control (HBAC) and the System Security Services Daemon (SSSD) to enable Active Directory users to log in to {Project}.

Author: Hm, not quite. I see configuring a trust as part of the actual FreeIPA installation; and, by analogy, it would feel weird to write "If you install FreeIPA". But I think this might address your concerns as well:

```suggestion
When your {FreeIPA} deployment includes a cross-forest trust with Active Directory, configure host-based access control (HBAC) and the System Security Services Daemon (SSSD) to enable Active Directory users to log in to {Project}.
```

> When a cross-forest trust is configured in your {FreeIPA} deployment, configure host-based access control (HBAC) and the System Security Services Daemon (SSSD) to enable Active Directory users to log in to {Project}.

Reviewer: Yeah, that works.

For more information, see link:{RHELDocsBaseURL}8/html-single/installing_identity_management/index#installing-trust-between-idm-and-ad_installing-identity-management[{RHEL}{nbsp}8 _Installing trust between IdM and AD_].

.Procedure

On your {FreeIPA} server:

. Enable HBAC:
.. Create an external group and add the AD group to it.
.. Add the new external group to a POSIX group.
.. Use the POSIX group in a HBAC rule.

On the {FreeIPA} server and all replicas in your {FreeIPA} topology, configure SSSD to transfer additional attributes of AD users:

. Add the AD user attributes to the _nss_ and _domain_ sections in `/etc/sssd/sssd.conf`.
For example:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
[domain/EXAMPLE.com]
...
krb5_store_password_if_offline = True
ldap_user_extra_attrs=email:mail, lastname:sn, firstname:givenname

[nss]
user_attributes=+email, +firstname, +lastname

[ifp]
allowed_uids = ipaapi, root
user_attributes=+email, +firstname, +lastname
----
. Clear the SSSD cache:
.. Stop SSSD:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
# systemctl stop sssd
----
. Clear the cache:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----

> # sss_cache -E

Author: https://issues.redhat.com/browse/SAT-23355 recommends manually deleting cached directories to clear the cache. I believe using sss_cache is the more appropriate choice.

----
.. Start SSSD:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
# systemctl start sssd
----

> . Verify the AD attributes value.
> Use the `dbus-send` command on your {ProjectServer} and on your {FreeIPA} server and make sure that both outputs match.

Reviewer: ```suggestion
. Verify the AD attributes value by comparing the output of the `dbus-send` command on your {ProjectServer} and on your {FreeIPA} server:
```

Author: I want to be explicit about the requirement for both outputs to match. I also find it helpful to deliver the information in two sentences to give users a bit of an extra breathing space. So I intend to go with this:

```suggestion
. Verify the AD attributes value by using the `dbus-send` command on your {ProjectServer} and on your {FreeIPA} server.
Make sure that both outputs match.
```

+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
[nss]

> user_attributes=+mail, +sn, +givenname

Reviewer 2: Shouldn't this stay the way it was?

Author: This is based on Waldirio's case notes in https://issues.redhat.com/browse/SAT-23355. He requests changing the NSS config line to use the attribute mappings configured in [ldap_user_extra_attrs](https://jhrozek.fedorapeople.org/sssd/1.11.90/man/sssd-ldap.5.html) and moving the [nss] section below the line with ldap_user_extra_attrs. It does seem to make sense based on the description in sssd-ldap(5) but I'm happy to change this if you have any concerns @adamruzicka
